src/test_l1_tvSettings.c

```diff
@@ -2765,21 +2833,23 @@ void test_l1_tvSettings_negative_SetCurrentBacklightMode (void)
 	UT_ASSERT_AUTO_TERM_NUMERICAL(result, tvERROR_INVALID_PARAM);
 
 	/* Step 06: Calling tvsettings SetCurrentBacklightMode and expecting the API to return tvERROR_INVALID_PARAM */
-	numberofBacklightModes = Configfile.backLightCtl.size;
+	numberofBacklightModes = UT_KVP_PROFILE_GET_LIST_COUNT("tvSettings/BacklightControl/index");
 	for(int i =0 ; i < numberofBacklightModes; i++)
 	{
 		SupportAvailable = false;
+		snprintf(keyValue, UT_KVP_MAX_ELEMENT_SIZE, "tvSettings/BacklightControl/index/%d", i);
+		modeValue = UT_KVP_PROFILE_GET_UINT32(keyValue);
 		for(int j = tvBacklightMode_NONE  ; j < tvBacklightMode_MAX ; j++)
 		{
-			if(Configfile.backLightCtl.modevalue[i] == j)
+			if(modeValue == j)
```

Comment:
this doesn't look correct.

All the fields must be supported, or return NOT SUPPORTED by calling the code.

So you have. tocall `SetCurrentBacklightMode() ` with every enum

Reply:
This is the same for all settings functions

Every API must Support Every Field and Every call.

Reply:
Need further discussion to tackle it in different work element.